automate-collections-updates/automate_collections_updates.py

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env python
+"""Update requirements.yml with latest collection versions."""
+
+import argparse
+import os
+import re
+import shutil
+import subprocess
+import sys
+import tempfile
+import yaml
+import tarfile
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--vendored_collections",
+        type=str,
+        default=os.environ.get("VENDORED_COLLECTIONS", "vendored_collections.yml"),
+        help="Path/filename for vendored_collections.yml",
+    )
+    parser.add_argument(
+        "--repository",
+        type=str,
+        default=os.environ.get("REPOSITORY", "rhel-system-roles"),
+        help="The repository to work on. Either rhel-system-roles or linux-system-roles",
+    )
+    args = parser.parse_args()
+
+    hsh = yaml.safe_load(open(args.vendored_collections))
+    collection_tarballs = {}
+
+    repo = args.repository
+    if repo == "rhel-system-roles":
+        repo_url = "https://src.osci.redhat.com:443/rpms/rhel-system-roles.git"
+        branch = "rhel-9.0.0"
+        pkg_command = "rhpkg"
+    elif repo == "linux-system-roles":
+        repo_url = "https://src.fedoraproject.org/rpms/linux-system-roles.git"
+        branch = "rawhide"
+        pkg_command = "fedpkg"
+    else:
+        sys.exit(
+            "repository must be set to either rhel-system-roles or linux-system-roles"
+        )
+
+    for coll in hsh["collections"]:
+        coll_install_dir = tempfile.mkdtemp()
+        coll_name = coll["name"]
+        coll_content_values = set()
+        for values in list(coll["content"].values()):
+            for value in values:
+                coll_content_values.add(value)
+        coll_dir = os.path.join(
+            coll_install_dir,
+            "ansible_collections",
+            coll_name.split(".")[0],
+            coll_name.split(".")[1],
+        )
+        cmd = [
+            "ansible-galaxy",
+            "collection",
+            "install",
+            "-n",
+            "-vv",
+            "--force",
+            "-p",
+            coll_install_dir,
+            coll_name,
+        ]
+        try:
+            """Get the latest version of the given collection"""
+            print(f"Installing the {coll_name} collection.")
+            out = subprocess.run(cmd, stdout=subprocess.PIPE, encoding="utf-8")
+            pat = f"Installing '{coll_name}:([^']+)' to"
+            coll_latest_ver = re.search(pat, out.stdout).group(1)
+            if coll_latest_ver == coll["version"]:
+                print(f"{coll_name} is of latest version. Skipping.")
+            else:
+                print(f"Searching {coll_name} changelog for updates.")
+                for value in coll_content_values:
+                    changelog = yaml.safe_load(
+                        open(os.path.join(coll_dir, "changelogs/changelog.yaml"))
+                    )
```

Comment:
do we need to look for updates in the changelog?  Can't we just assume if the version is different, it is newer, and use it?  I guess that would cause problems if they updated the version a lot for trivial, non-API issues, but I don't think that is the case.

Reply:
I pushed this as a suggestion for you to look at. And now with your comment I understand that it is a loss of time to check for the changelog. It is not a big deal for us to vendor a new version if it got released. I'll remove this logic.